tests/test_citus.py

```diff
@@ -0,0 +1,159 @@
+            mock_query.assert_called_once()
+            self.assertEqual(mock_query.call_args[0][0], 'COMMIT')
+
+    def test_process_tasks(self):
+        self.c.add_task('after_promote', 0, 'postgres://host2:5432/postgres')
+        self.c.process_tasks()
+
+        self.c.add_task('after_promote', 0, 'postgres://host3:5432/postgres')
+        with patch('patroni.postgresql.citus.logger.error') as mock_logger,\
+                patch.object(CitusHandler, 'query', Mock(side_effect=Exception)):
+            self.c.process_tasks()
+            mock_logger.assert_called_once()
+            self.assertTrue(mock_logger.call_args[0][0].startswith('Exception when working with pg_dist_node: '))
+
+    def test_on_demote(self):
+        self.c.on_demote()
+
+    @patch('patroni.postgresql.citus.logger.error')
+    @patch.object(MockCursor, 'execute', Mock(side_effect=Exception))
+    def test_load_pg_dist_node(self, mock_logger):
+        # load_pg_dist_node() triggers, query fails and exception is property handled
+        self.c.process_tasks()
+        self.assertTrue(self.c._schedule_load_pg_dist_node)
+        mock_logger.assert_called_once()
+        self.assertTrue(mock_logger.call_args[0][0].startswith('Exception when executing query'))
+        self.assertTrue(mock_logger.call_args[0][1].startswith('SELECT nodeid, groupid, '))
+
+    def test_wait(self):
+        task = self.c.add_task('before_demote', 1, 'postgres://host:5432/postgres', 30)
+        task._event.wait = Mock()
+        task.wait()
+
+    def test_adjust_postgres_gucs(self):
+        parameters = {'max_connections': 101,
+                      'max_prepared_transactions': 0,
+                      'shared_preload_libraries': 'foo , citus, bar '}
+        self.c.adjust_postgres_gucs(parameters)
+        self.assertEqual(parameters['max_prepared_transactions'], 202)
+        self.assertEqual(parameters['shared_preload_libraries'], 'citus,foo,bar')
+        self.assertEqual(parameters['wal_level'], 'logical')
+
+    @patch.object(CitusHandler, 'is_enabled', Mock(return_value=False))
+    def test_bootstrap(self):
+        self.c.bootstrap()
+
+    def test_ignore_replication_slot(self):
+        self.assertFalse(self.c.ignore_replication_slot({'name': 'foo', 'type': 'physical',
+                                                         'database': 'bar', 'plugin': 'wal2json'}))
+        self.assertFalse(self.c.ignore_replication_slot({'name': 'foo', 'type': 'logical',
+                                                         'database': 'bar', 'plugin': 'wal2json'}))
+        self.assertFalse(self.c.ignore_replication_slot({'name': 'foo', 'type': 'logical',
+                                                         'database': 'bar', 'plugin': 'pgoutput'}))
+        self.assertFalse(self.c.ignore_replication_slot({'name': 'foo', 'type': 'logical',
+                                                         'database': 'citus', 'plugin': 'pgoutput'}))
```

Comment:
> 'database': 'citus'

I more and more think about having such values used in different tests stored in dedicated variables   🤔